app/src/main/java/unipd/se18/ocrcamera/IngredientsActivity.java

```diff
@@ -0,0 +1,81 @@
+package unipd.se18.ocrcamera;
+
+import android.app.ProgressDialog;
+import android.content.SharedPreferences;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.widget.ListView;
+
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+
+/**
+ * Activity that shows informations taken from inci db about the ingredients found in the ocr text
+ * @author Francesco Pham
+ */
+public class IngredientsActivity extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_ingredients);
+
+        //get ocr text
+        SharedPreferences prefs = getSharedPreferences("prefs", MODE_PRIVATE);
+        String OCRText = prefs.getString("text", null);
+
+        if(OCRText != null) {
+            InciThread inciThread = new InciThread(OCRText);
+            inciThread.start();
+        }
+    }
+
+    class InciThread extends Thread{
```

Comment:
Why are you using a thread here? couldn't you run this piece of code directly on the main thread?

Reply:
Because the processing takes a long time we can't stop the main UI thread, it would crash the app.

Reply:
I mean, if you are comparing 'aqua' with 'gas' (two words that have nothing in common) can we stop the evaluation of the levenshtein distance before (for example noticing that the distance is greater than 3) and save some time? This reasoning could be done for many other ingredients